doc/introduction/measurements.rst

```diff
@@ -355,15 +364,35 @@ Currently, statistics can only be collected for single mid-circuit measurement v
 measurement values manipulated using boolean or arithmetic operators cannot be used. These can lead to
 unexpected/incorrect behaviour.
```

Comment:
Reading through the docs for this section feels like receiving a slightly unstructured stream of consciousness. We need to have some subsections, e.g.:
- basics
- conditioning and postselection
- statistics
- deferred measurement
Would we have capacity to do this in a follow up PR before the release?

Reply:
Since I still need two reviews, I can add the changes to this PR. Shouldn't block anything.